src/graphs/hld.rs

```diff
@@ -0,0 +1,130 @@
+//! # Heavy Light Decomposition
+
+use crate::graphs::dfs_order::{get_dfs_postorder, get_dfs_preorder};
+use std::ops::Range;
+
+/// # Example
+/// ```
+/// use programming_team_code_rust::graphs::hld::HLD;
+/// use programming_team_code_rust::data_structures::fenwick::Fenwick;
+///
+/// let mut adj = vec![
+///    vec![1, 2],
+///    vec![0, 3, 4],
+///    vec![0, 5],
+///    vec![1],
+///    vec![1],
+///    vec![2],
+/// ];
+/// let mut fenwick = Fenwick::<usize>::build_on_array(&[3, 2, 4, 5, 1, 2]);
+///
+/// let hld = HLD::new(&mut adj, false);
+/// assert_eq!(hld.lca(3, 4), 1);
+/// assert_eq!(fenwick.sum(hld.sub_tree(0)), 17);
+/// let mut sum = 0;
+/// hld.path(3, 4, |range, _| sum += fenwick.sum(range));
+/// assert_eq!(sum, 11);
```

Comment:
this is my idea for the example: only one function call to each member function

basically my goal is mainly to show the syntax